### PR TITLE
Fixed bugs for 1D array inputs on `vstack` , `dstack` and `column_stack`

### PR DESCRIPTION
**Pull request by sbak5**
Found bugs for 1D array inputs 
because reshaping of this input array didn't happen before checking the shape of arrays in input array list.

### CODE REVIEW

File: tests/concatenate_stack.py [hunk 1]
```diff
@@ -65,28 +68,37 @@ def run_test(arr, routine, input_size):
...
-                for num_arr in range(3)
-            ]
-        )
-        run_test(a, routine, input_size)
+        a = [
+            np.random.randint(low=0, high=100, size=(input_size))
+            for num_arr in range(3)
+        ]
+        # test the exception for 1D array on vstack and dstack
+        if routine == "vstack" or routine == "dstack":
```
Comment (magnatelee) [inline tests/concatenate_stack.py:96]: a nitpick: you can shorten this to `if routine in ("vstack", "dstack"):`.

File: cunumeric/module.py [hunk 2]
```diff
@@ -1261,8 +1267,9 @@ def _concatenate(
     # Check to see if we can build a new tuple of cuNumeric arrays
     leading_dim = 0
     cunumeric_inputs = inputs
-
-    leading_dim = common_info.shape[axis] * len(cunumeric_inputs)
+    for arr in cunumeric_inputs:
+        leading_dim += arr.shape[axis]
```
Comment (magnatelee) [inline cunumeric/module.py:1271]: can this be rewritten to `leading_dim = sum(arr.shape[axis] for arr in inputs)`?

File: cunumeric/module.py [hunk 3]
```diff
@@ -1524,17 +1532,19 @@ def dstack(tup):
...
-        if common_info.ndim == 1:
-            shape.insert(0, 1)
-        shape.append(1)
-        common_info.shape = shape
-        for i, arr in enumerate(tup):
-            tup[i] = arr.reshape(shape)
+    reshaped = []
+    inputs = list(ndarray.convert_to_cunumeric_ndarray(inp) for inp in tup)
+    for arr in inputs:
+        if arr.ndim <= 2:
```
Comment (magnatelee) [inline cunumeric/module.py:1539]: A suggestion:
```
if arr.ndim == 1:
  arr = arr.reshape((1,) + arr.shape + (1,))
elif arr.ndim == 2:
  arr = arr.reshape(arr.shape + (1,))
```
Nested conditions in this if block seem to make the code unnecessarily obscure.

File: cunumeric/module.py (hunk 2 above)
Comment (sbak5) [reply cunumeric/module.py:1271]: I also think your code w/ generator should work but on prom, `leading_dim` becomes generator for some reason, which makes the ndarray creation fail. 
`leading_dim = sum(list(arr.shape[axis] for arr in inputs))` works w/o the error. On my local machine, your code works.

Comment (magnatelee) [reply cunumeric/module.py:1271]: What Python version are you using on prom?

Comment (sbak5) [reply cunumeric/module.py:1271]: It is 3.8.12

Comment (magnatelee) [reply cunumeric/module.py:1271]: btw, I think the symbol `sum` being redefined in `module.py` can be problematic later on, so maybe you should make a copy of the builtin sum at the beginning before it gets redefined, just like how some of other builtins are handled here: https://github.com/nv-legate/cunumeric/blob/branch-22.03/cunumeric/module.py#L31-L35

Comment (magnatelee) [reply cunumeric/module.py:1271]: can you try the workaround for the builtin sum I mentioned and see if that changes anything?

Comment (sbak5) [reply cunumeric/module.py:1271]: It worked. the sum implementation in `module.py` was used for the statement.

Comment (magnatelee) [reply cunumeric/module.py:1271]: yeah that's what I thought. a similar thing tripped me over the other day and I learned it the hard way.

Comment (sbak5) [reply cunumeric/module.py:1271]: Then, I'll add the _builtin_sum code in `module.py` Thanks for saving me out of this issue :)

**Comment by magnatelee** [comment]
LGTM. Just reran the CI to make sure all tests pass. (there was an intermittent issue with the CI yesterday.)
